pages/admin/settings/components/BusinessForm.tsx

```diff
@@ -0,0 +1,195 @@
+import {
+    Box,
+    VStack,
+    Text,
+    Flex,
+    Button,
+    Spinner,
+    Input,
+    Center,
+    Skeleton,
+} from '@chakra-ui/react';
+import React, { useEffect, useState } from 'react';
+import { updateUserProfile } from '@/services/userProfileService';
+import {
+    BusinessFormData,
+    hasBusinessFormChanges,
+    showBusinessNotification,
+    validateBusinessFields,
+} from '@/utils/userProfile/userProfileValidation';
+import { BusinessInfo } from '@/context/BusinessContext';
+
+interface BusinessFormProps {
+    businessInfo: BusinessInfo;
+    setBusinessInfo: (info: BusinessInfo) => void;
+    isSmallScreen: boolean;
+    isLoading: boolean;
+}
+
+export default function BusinessForm({
+    businessInfo,
+    setBusinessInfo,
+    isSmallScreen,
+    isLoading,
+}: BusinessFormProps) {
+    const [name, setName] = useState('');
+    const [address, setAddress] = useState('');
+    const [phone, setPhone] = useState('');
+    const [instagram, setInstagram] = useState('');
+    const [isSaving, setIsSaving] = useState(false);
+
+    useEffect(() => {
+        setName(businessInfo.name);
```

Comment:
💭 why having different states for each property rather than building a single one with all of the properties? 🤔